stdlib/LinearAlgebra/src/blas.jl

```diff
@@ -423,7 +423,11 @@ nrm2(x::Union{AbstractVector,DenseArray}) = nrm2(length(x), x, stride1(x))
 """
     asum(n, X, incx)
 
-Sum of the absolute values of the first `n` elements of array `X` with stride `incx`.
+Sum of the magnitudes of the first `n` elements of array `X` with stride `incx`.
+
+For a real array, the magnitude is the absolute value. For a complex array, the
+magnitude is the sum of the absolute value of the real part and the absolute value
+of the imaginary part.
```

Comment:
We should make it clear that this is not the normal definition of magnitude for a complex number.
Maybe we should say the `L1 norm` instead?

Reply:
I think the proposed formulation is pretty clear. Maybe we could mention explicitly that no square roots are taken but I don't really think it's required. I assume you mean `ℓ1`-norm of each element but I think it would just be unnecessarily confusing to use functional analysis language for applying `abs2` to the elements of a complex array.